src/huggingface_hub/utils/cache_explorer.py

```diff
@@ -0,0 +1,246 @@
+                )
+
+            blob_path = Path(file_path).resolve()
+            if not blob_path.exists():
+                raise CorruptedCacheException("Blob missing (broken symlink) !")
+
+            if blobs_path not in blob_path.parents:
+                raise CorruptedCacheException(
+                    "Blob symlink points outside of blob directory !"
+                )
+
+            if blob_path not in blob_sizes:
+                blob_sizes[blob_path] = blob_path.stat().st_size
+
+            cached_files.append(
+                CachedFileInfo(
+                    file_name=file_path.name,
+                    file_path=file_path,
+                    size_on_disk=blob_sizes[blob_path],
+                    blob_path=blob_path,
+                )
+            )
+
+        cached_revisions_by_hash[revision_path.name] = CachedRevisionInfo(
+            commit_hash=revision_path.name,
+            size_on_disk=sum(
+                blob_sizes[blob_path]
+                for blob_path in set(file.blob_path for file in cached_files)
+            ),
+            files=sorted(cached_files, key=lambda file: file.file_path),
+        )
+
+    # Scan over `refs` directory
+    if refs_path.exists():
+        # Example of `refs` directory
+        # ── refs
+        #     ├── main
+        #     └── refs
+        #         └── pr
+        #             └── 1
+        if refs_path.is_file():
+            raise CorruptedCacheException("Refs directory cannot be a file !")
+
+        for ref_path in refs_path.glob("**/*"):
+            # glob("**/*") iterates over all files and directories -> skip directories
+            if ref_path.is_dir():
+                continue
+
+            ref_name = str(ref_path.relative_to(refs_path))
+            with ref_path.open() as f:
+                commit_hash = f.read()
+
+            if commit_hash not in cached_revisions_by_hash:
+                raise CorruptedCacheException(
+                    "Reference refers to a missing commit hash !"
+                )
+
+            cached_revision = cached_revisions_by_hash[commit_hash]
+            if cached_revision.refs is None:
+                cached_revision.refs = {ref_name}
+            else:
+                cached_revision.refs.add(ref_name)
+
+    return CachedRepoInfo(
+        repo_id=repo_id,
+        repo_type=repo_type,
+        revisions=sorted(
+            cached_revisions_by_hash.values(), key=lambda revision: revision.commit_hash
+        ),
+        repo_path=repo_path,
+        size_on_disk=sum(blob_sizes.values()),
+        nb_files=len(blob_sizes),
+    )
+
+
+def _format_size(num: int, suffix: str = "B") -> str:
+    """Format size in bytes into a human-readable string.
+
+    Taken from https://stackoverflow.com/a/1094933
+    """
+    for unit in ["", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"]:
+        if abs(num) < 1024.0:
+            return f"{num:3.1f}{unit}{suffix}"
+        num /= 1024.0
+    return f"{num:.1f}Yi{suffix}"
```

Comment:
personally i think now most people stick to SI units ie. powers of 10 not of 2

aka `MB` not `MiB`

It also makes the code simpler :)

WDYT?

Reply:
(@Pierrci for this too)

Reply:
also consistent with the Hub where we display GB not GiB: https://huggingface.co/gpt2/tree/main

Reply:
yes, +1 on consistency and using GB instead of GiB :)

Reply:
my OS (which is the superior OS™️) also displays GB... I think at this point `GiB` are for dinosaurs =)

Reply:
Yep, will update this. Haven't thought too much about it to be honest 😁